### PR TITLE
Requisition Centralization [Part 1.5] AKA medvendor infinite restock outside medbay cheese removal redux

### PR DESCRIPTION
**Pull request by ItsVyzo**
# About the pull request

This PR adds a new structure: the medical supply link port. This is a structure that allows medical vendors to link to the Almayer's internal medical supplies and have larger restocking capabilities. They can be found around the Almayer where existing medical vendors are.

This PR makes it impossible to restock/refill non full stacks at medical vendors which are not linked to a medical supply link port

This PR makes it impossible to restock/refill non full autoinjectors and bottles at medical vendors which are not linked to a medical supply link port

revives #2155

# Explain why it's good for the game

This PR is good for the game because it will get req more involved in the medical supply process, as well as limiting the capability of medics to instantly resupply from a vendor at the fob or in hydro. This makes it so that medics have to more carefully pick their supplies, and plan ahead for resupplying, as well as consider other medics' use of supplies, as nothing is infinite groundside anymore.

Overall, it will make the medic supply gameplay loop more engaging and interesting, as well as reducing marine capability to infinitely heal and resupply themselves groundside.

Makes #5490 more useable

# Testing Photographs and Procedure

<details>

https://gyazo.com/d9e8e3b63daab1241e046c23bb5dd140

</details>

# Changelog

🆑 Stan_albatross,kugamo,LTNTS,Drathek
balance: medical vendors, excepting nanomeds, can now only refill/restock stacks and autoinjectors/bottles when on special medical supply link ports that can only be found shipside. Sprite by kugamo.
fix: supply pads will no longer generate ambient occlusion.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+	name = "medical supplies link port"
+	desc = "A complex network of pipes and machinery, linking to large storage systems below the deck. Medical vendors linked to this port will be able to infinitely restock supplies."
+	icon = 'icons/effects/warning_stripes.dmi'
+	icon_state = "medlink_unclamped"
+	anchored = TRUE
+	density = FALSE
+	unslashable = TRUE
+	unacidable = TRUE
+	plane = FLOOR_PLANE
+	layer = 2.1 //It's the floor, man
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:13]: layers need defines

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 2]
```diff
@@ -1,3 +1,47 @@
+//------------SUPPLY LINK FOR MEDICAL VENDORS---------------
+
+/obj/structure/medical_supply_link
+	name = "medical supplies link port"
+	desc = "A complex network of pipes and machinery, linking to large storage systems below the deck. Medical vendors linked to this port will be able to infinitely restock supplies."
+	icon = 'icons/effects/warning_stripes.dmi'
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:6]: is this correct?

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 3]
```diff
@@ -1,3 +1,47 @@
...
+	UnregisterSignal(src, COMSIG_STRUCTURE_WRENCHED)
+	UnregisterSignal(src, COMSIG_STRUCTURE_UNWRENCHED)
+	return ..()
+
+/obj/structure/medical_supply_link/proc/do_clamp_animation()
+	flick("medlink_clamping", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/proc/do_unclamp_animation()
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:34]: doc comments please

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+
+/obj/structure/medical_supply_link/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_STRUCTURE_WRENCHED, PROC_REF(do_clamp_animation))
+	RegisterSignal(src, COMSIG_STRUCTURE_UNWRENCHED, PROC_REF(do_unclamp_animation))
+	update_icon()
+
+/obj/structure/medical_supply_link/deconstruct(disassembled)
+	UnregisterSignal(src, COMSIG_STRUCTURE_WRENCHED)
+	UnregisterSignal(src, COMSIG_STRUCTURE_UNWRENCHED)
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:26]: not needed

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 5]
```diff
@@ -14,6 +58,8 @@
 	vendor_theme = VENDOR_THEME_COMPANY
 	vend_delay = 0.5 SECONDS
 
+	var/requires_supply_link_port = TRUE
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:61]: doc comment

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 6]
```diff
@@ -60,6 +106,33 @@
...
+					playsound(src, 'sound/machines/buzz-sigh.ogg', 15, TRUE)
+				return FALSE
+
+	//stacked items handling if the vendor cannot restock partial stacks
+	else if(istype(item_to_stock, /obj/item/stack))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/stack/restock_stack = item_to_stock
+			if(restock_stack.amount < restock_stack.max_amount) // if the stack is not full
+				if(user)
+					to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects \the [restock_stack]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:131]: \the not needed

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 7]
```diff
@@ -60,6 +106,33 @@
...
+		return FALSE
+	return TRUE
+
+/obj/structure/machinery/cm_vending/sorted/medical/additional_restock_checks(obj/item/item_to_stock, mob/user)
+	if(istype(item_to_stock, /obj/item/reagent_container/hypospray/autoinjector) || istype(item_to_stock, /obj/item/reagent_container/glass/bottle))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/reagent_container/container = item_to_stock
+			if(container.reagents.total_volume < container.reagents.maximum_volume)
+				if(user)
+					to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects \the [container.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:121]: ditto

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 8]
```diff
@@ -60,6 +106,33 @@
 	if(healthscan)
 		. += SPAN_NOTICE("The [src.name] offers assisted medical scan, for ease of usage with minimal training. Present the target in front of the scanner to scan.")
 
+/obj/structure/machinery/cm_vending/sorted/medical/proc/get_supply_link()
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:109]: doc

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 9]
```diff
@@ -80,6 +153,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [C.name] is currently full."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects \the [C.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:157]: \the

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 10]
```diff
@@ -104,6 +182,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [S.name] is currently fully stacked."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects \the [S.name]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/medical.dm:186]: \the

File: code/game/objects/structures.dm [hunk 11]
```diff
@@ -209,7 +209,7 @@
 
 /obj/structure/proc/toggle_anchored(obj/item/W, mob/user)
 	if(!wrenchable)
-		to_chat(user, SPAN_WARNING("[src] cannot be [anchored ? "un" : ""]anchored."))
+		to_chat(user, SPAN_WARNING("The [src] cannot be [anchored ? "un" : ""]anchored."))
```
Comment (Zonespace27) [inline code/game/objects/structures.dm:212]: Why did you add this

File: code/game/objects/structures.dm [hunk 12]
```diff
@@ -219,8 +219,12 @@
 			playsound(loc, 'sound/items/Ratchet.ogg', 25, 1)
 			if(anchored)
 				user.visible_message(SPAN_NOTICE("[user] anchors [src] into place."),SPAN_NOTICE("You anchor [src] into place."))
+				for(var/obj/O in loc)
```
Comment (Zonespace27) [inline code/game/objects/structures.dm:222]: no one-letter vars

File: code/game/objects/structures.dm [hunk 13]
```diff
@@ -219,8 +219,12 @@
...
+					SEND_SIGNAL(O, COMSIG_STRUCTURE_WRENCHED, src)
 			else
 				user.visible_message(SPAN_NOTICE("[user] unanchors [src]."),SPAN_NOTICE("You unanchor [src]."))
+				for(var/obj/O in loc)
```
Comment (Zonespace27) [inline code/game/objects/structures.dm:226]: no one-letter vars

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures.dm (hunk 11 above)
Comment (ItsVyzo) [reply code/game/objects/structures.dm:212]: oversight

File: code/game/machinery/vending/vendor_types/medical.dm (hunk 2 above)
Comment (ItsVyzo) [reply code/game/machinery/vending/vendor_types/medical.dm:6]: Do you mean grammar, icon or desc?

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Huffie56** [comment]
from what i can see map wise you changed the floor tiles for no reason?
image
shouldn't be full white when it's close to a wall but half green (part toward the wall) and the rest white...
look at the other vendors around them...

**Comment by ItsVyzo** [comment]
> from what i can see map wise you changed the floor tiles for no reason?
> 
> image
> 
> shouldn't be full white when it's close to a wall but half green (part toward the wall) and the rest white...
> 
> look at the other vendors around them...

Floor was not changed. That's a medlink under the vendors and it's not green to allow distinguishing from afar

**Review by Drulikar — DISMISSED** [review]
What this PR is doing is good; however I don't want to entirely disable refilling of injectors if not in a port. Rather the vendor should have a pool of reagents that is consumed as autoinjectors are filled.

Additionally I would like the vendor to have a more streamlined means to restock it; e.g. a vendor restock pack that req can order and then can be used on a vendor to restore the stock values (if any need stocking).

Reagents can be a separate restock pack (instead of also being refilled via the supply pack); but ultimately its the same process. I don't want to go so far as trying to get chemistry to actually mix together the reagents but ever chemistry needs more work that could be something for them to do.

So in summary:
- Vendors not in a restock port need to still allow injector refilling, but this pulls from either a reagent amount per type of chemical - or a omni reagent amount that can be used for any chemical.
- Vendors not in a restock port need stocking item(s) that req can order and then be used on a vendor to restock it. Its fine to allow individual items to be inserted into a vendor, but this shouldn't be the only way to restock it.

If these two things are a bit more difficult for you to implement, we can shelve this PR until another PR is put together to address those requests.

**Comment by Drulikar** [comment]
Also we need to disallow splitting of stacks of items that we can top off (e.g. so you can't vend one trauma kit, split it, top it off, and now have duplicated the stack).

**Comment by Drulikar** [comment]
Stale exempt because its roadmap - and waiting on me anyways

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: icons/effects/warning_stripes.dmi
Comment (Nanu308) [inline icons/effects/warning_stripes.dmi:1]: bilde
bilde

Why is the turf a part of the structure sprite? This makes the map stuff look really bad in some places.

I want this changed for map aesthetics.

**Review by Nanu308 — CHANGES_REQUESTED** [review]

File: icons/effects/warning_stripes.dmi
Comment (ItsVyzo) [reply icons/effects/warning_stripes.dmi:1]: will work on that soon-ish

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: icons/effects/warning_stripes.dmi
Comment (ItsVyzo) [reply icons/effects/warning_stripes.dmi:1]: resolvedd

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 14]
```diff
@@ -1,3 +1,103 @@
...
+	else
+		icon_state = "medlink_unclamped"
+
+// --- No Upper Tile / White
+/obj/structure/medical_supply_link/notop
+	icon_state = "medlink_unclamped_notop"
+
+/obj/structure/medical_supply_link/notop/ex_act(severity, direction)
+	return FALSE
+
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:49]:
```suggestion
```
Redundant to parent's implementation.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 15]
```diff
@@ -1,3 +1,103 @@
+//------------SUPPLY LINK FOR MEDICAL VENDORS---------------
+
+/obj/structure/medical_supply_link
+	name = "medilink supply port"
+	desc = "A complex network of pipes and machinery, linking to large storage systems below the deck. Medical vendors linked to this port will be able to infinitely restock supplies."
+	icon = 'icons/effects/warning_stripes.dmi'
+	icon_state = "medlink_unclamped"
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:7]:
```suggestion
	icon_state = "medlink_unclamped"
	var/base_state = "medlink"
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 16]
```diff
@@ -1,3 +1,103 @@
...
+
+/obj/structure/medical_supply_link/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_STRUCTURE_WRENCHED, PROC_REF(do_clamp_animation))
+	RegisterSignal(src, COMSIG_STRUCTURE_UNWRENCHED, PROC_REF(do_unclamp_animation))
+	update_icon()
+
+/obj/structure/medical_supply_link/deconstruct(disassembled)
+	return ..()
+
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:26]:
```suggestion
```
You aren't doing anything with the proc so what's the point.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 17]
```diff
@@ -1,3 +1,103 @@
...
+/obj/structure/medical_supply_link/proc/do_unclamp_animation()
+	flick("medlink_unclamping", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/update_icon()
+	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
+		icon_state = "medlink_clamped"
+	else
+		icon_state = "medlink_unclamped"
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:41]:
```suggestion
/obj/structure/medical_supply_link/proc/do_clamp_animation() // clamp and unclamp animations for when vendor is wrench over supply link
	flick("[base_state]_clamping", src)
	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
	update_icon()

/obj/structure/medical_supply_link/proc/do_unclamp_animation()
	flick("[base_state]_unclamping", src)
	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
	update_icon()

/obj/structure/medical_supply_link/update_icon()
	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
		icon_state = "[base_state]_clamped"
	else
		icon_state = "[base_state]_unclamped"
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 18]
```diff
@@ -1,3 +1,103 @@
...
+
+// --- No Upper Tile / White
+/obj/structure/medical_supply_link/notop
+	icon_state = "medlink_unclamped_notop"
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:45]:
```suggestion
	icon_state = "medlink_notop_unclamped"
	base_state = "medlink_notop"
```
**NOTE:** You must also update the DMI to accommodate this. We need to have the unclamped and clamped be the suffix.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 19]
```diff
@@ -1,3 +1,103 @@
...
+	flick("medlink_unclamping_notop", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/notop/update_icon()
+	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
+		icon_state = "medlink_clamped_notop"
+	else
+		icon_state = "medlink_unclamped_notop"
+
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:71]:
```suggestion
```
This is all redundant if you do the base_state changes.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 20]
```diff
@@ -1,3 +1,103 @@
...
+// --- No Upper Tile / Green
+/obj/structure/medical_supply_link/notopg
+	icon_state = "medlink_unclamped_notopg"
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:74]:
```suggestion
	icon_state = "medlink_notopg_unclamped"
	base_state = "medlink_notopg"
```
**NOTE**: You must also update the DMI to accommodate this. We need to have the unclamped and clamped be the suffix.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 21]
```diff
@@ -1,3 +1,103 @@
...
+
+/obj/structure/medical_supply_link/notopg/ex_act(severity, direction)
+	return FALSE
+
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:78]:
```suggestion
```
Redundant to parent's implementation.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 22]
```diff
@@ -1,3 +1,103 @@
...
+	flick("medlink_unclamping_notopg", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/notopg/update_icon()
+	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
+		icon_state = "medlink_clamped_notopg"
+	else
+		icon_state = "medlink_unclamped_notopg"
+
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:100]:
```suggestion
```
This is all redundant if you do the base_state changes.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 23]
```diff
@@ -60,6 +162,33 @@
...
+		return FALSE
+	return TRUE
+
+/obj/structure/machinery/cm_vending/sorted/medical/additional_restock_checks(obj/item/item_to_stock, mob/user)
+	if(istype(item_to_stock, /obj/item/reagent_container/hypospray/autoinjector) || istype(item_to_stock, /obj/item/reagent_container/glass/bottle))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/reagent_container/container = item_to_stock
+			if(container.reagents.total_volume < container.reagents.maximum_volume)
+				if(user)
+					to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects [container.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:177]:
```suggestion
					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [container.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 24]
```diff
@@ -60,6 +162,33 @@
...
+					playsound(src, 'sound/machines/buzz-sigh.ogg', 15, TRUE)
+				return FALSE
+
+	//stacked items handling if the vendor cannot restock partial stacks
+	else if(istype(item_to_stock, /obj/item/stack))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/stack/restock_stack = item_to_stock
+			if(restock_stack.amount < restock_stack.max_amount) // if the stack is not full
+				if(user)
+					to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects [restock_stack]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:187]:
```suggestion
					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [restock_stack]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 25]
```diff
@@ -80,6 +209,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [C.name] is currently full."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects [C.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:213]:
```suggestion
			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [C.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 26]
```diff
@@ -104,6 +238,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [S.name] is currently fully stacked."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("\The [src] makes a buzzing noise as it rejects [S.name]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:242]:
```suggestion
			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [S.name]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
Be sure to update the DMIs and test changes.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 27]
```diff
@@ -1,3 +1,48 @@
...
+	update_icon()
+
+/obj/structure/medical_supply_link/update_icon()
+	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
+		icon_state = "[base_state]_clamped"
+	else
+		icon_state = "[base_state]_unclamped"
+
+// --- No Upper Tile / Green
+/obj/structure/medical_supply_link/notopg
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:42]:
```suggestion
// --- Green
/obj/structure/medical_supply_link/green
```
Please also repath it in the map too.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 28]
```diff
@@ -1,3 +1,48 @@
...
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/proc/do_unclamp_animation()
+	flick("[base_state]_unclamping", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/update_icon()
+	if(locate(/obj/structure/machinery/cm_vending/sorted/medical) in loc)
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:36]:
```suggestion
	var/obj/structure/machinery/cm_vending/sorted/medical/vendor = locate() in loc
	if(vendor && vendor.anchored)
```
Without this the link basically always goes to the clamped state.

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 29]
```diff
@@ -60,6 +107,33 @@
 	if(healthscan)
 		. += SPAN_NOTICE("The [src.name] offers assisted medical scan, for ease of usage with minimal training. Present the target in front of the scanner to scan.")
 
+/obj/structure/machinery/cm_vending/sorted/medical/proc/get_supply_link() // checks supply link location in relation to medvendor
+	var/linkpoint = locate(/obj/structure/medical_supply_link) in loc
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:111]:
```suggestion
	if(!anchored)
		return FALSE
	var/obj/structure/medical_supply_link/linkpoint = locate() in loc
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 30]
```diff
@@ -60,6 +107,33 @@
...
+		return FALSE
+	return TRUE
+
+/obj/structure/machinery/cm_vending/sorted/medical/additional_restock_checks(obj/item/item_to_stock, mob/user)
+	if(istype(item_to_stock, /obj/item/reagent_container/hypospray/autoinjector) || istype(item_to_stock, /obj/item/reagent_container/glass/bottle))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/reagent_container/container = item_to_stock
+			if(container.reagents.total_volume < container.reagents.maximum_volume)
+				if(user)
+					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [container.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:122]:
```suggestion
					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [container.name]. Looks like this vendor cannot refill these without a connected supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 31]
```diff
@@ -60,6 +107,33 @@
...
+					playsound(src, 'sound/machines/buzz-sigh.ogg', 15, TRUE)
+				return FALSE
+
+	//stacked items handling if the vendor cannot restock partial stacks
+	else if(istype(item_to_stock, /obj/item/stack))
+		if(requires_supply_link_port && !get_supply_link())
+			var/obj/item/stack/restock_stack = item_to_stock
+			if(restock_stack.amount < restock_stack.max_amount) // if the stack is not full
+				if(user)
+					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [restock_stack]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:132]:
```suggestion
					to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [restock_stack]. Looks like this vendor cannot restock these without a connected supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 32]
```diff
@@ -80,6 +154,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [C.name] is currently full."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [C.name]. Looks like this vendor cannot refill these outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:158]:
```suggestion
			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [C.name]. Looks like this vendor cannot refill these without a connected supply link."))
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 33]
```diff
@@ -104,6 +183,11 @@
 			to_chat(user, SPAN_WARNING("[src] makes a warning noise. The [S.name] is currently fully stacked."))
 			return
 
+		if(requires_supply_link_port && !get_supply_link())
+			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [S.name]. Looks like this vendor cannot restock non-full stacks outside of a medical bay's supply link."))
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:187]:
```suggestion
			to_chat(user, SPAN_WARNING("[src] makes a buzzing noise as it rejects [S.name]. Looks like this vendor cannot restock these without a connected supply link."))
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 34]
```diff
@@ -1,3 +1,48 @@
...
+/obj/structure/medical_supply_link/ex_act(severity, direction)
+	return FALSE
+
+/obj/structure/medical_supply_link/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_STRUCTURE_WRENCHED, PROC_REF(do_clamp_animation))
+	RegisterSignal(src, COMSIG_STRUCTURE_UNWRENCHED, PROC_REF(do_unclamp_animation))
+	update_icon()
+
+/obj/structure/medical_supply_link/proc/do_clamp_animation() // clamp and unclamp animations for when vendor is wrench over supply link
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:25]:
```suggestion
/// Performs the clamping animation when a structure is anchored in our loc
/obj/structure/medical_supply_link/proc/do_clamp_animation()
	SIGNAL_HANDLER
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 35]
```diff
@@ -1,3 +1,48 @@
...
+	flick("[base_state]_clamping", src)
+	addtimer(CALLBACK(src, PROC_REF(update_icon), 2.6 SECONDS))
+	update_icon()
+
+/obj/structure/medical_supply_link/proc/do_unclamp_animation()
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:30]:
```suggestion
/// Performs the unclamping animation when a structure is unanchored in our loc
/obj/structure/medical_supply_link/proc/do_unclamp_animation()
	SIGNAL_HANDLER
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 36]
```diff
@@ -14,6 +59,8 @@
 	vendor_theme = VENDOR_THEME_COMPANY
 	vend_delay = 0.5 SECONDS
 
+	var/requires_supply_link_port = TRUE // sets vendor to require medlink to be able to resupply
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:62]:
```suggestion
	/// sets vendor to require a medlink to be able to resupply
	var/requires_supply_link_port = TRUE
```

File: code/game/machinery/vending/vendor_types/medical.dm [hunk 37]
```diff
@@ -60,6 +107,33 @@
 	if(healthscan)
 		. += SPAN_NOTICE("The [src.name] offers assisted medical scan, for ease of usage with minimal training. Present the target in front of the scanner to scan.")
 
+/obj/structure/machinery/cm_vending/sorted/medical/proc/get_supply_link() // checks supply link location in relation to medvendor
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/medical.dm:110]:
```suggestion
/// checks if there is a supply link in our location and we are anchored to it
/obj/structure/machinery/cm_vending/sorted/medical/proc/get_supply_link()
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
Outstanding issues not resolved from zonespace review.

**Review by Drulikar — APPROVED** [review]

**Review by sleepynecrons — APPROVED** [review]
sprites fine

File: maps/map_files/USS_Almayer/USS_Almayer.dmm
Comment (Nanu308) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:1]: mapping looking great with transparent version 👍

**Review by Nanu308 — APPROVED** [review]
